Source/SwiftLintFramework/Rules/Lint/NonPrivateXCTestMembersRule.swift

```diff
@@ -0,0 +1,69 @@
+import Foundation
+import SourceKittenFramework
+
+public struct NonPrivateXCTestMembersRule: Rule, OptInRule, ConfigurationProviderRule, AutomaticTestableRule,
+                                           SubstitutionCorrectableRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "non_private_xctest_member",
+        name: "Non-private XCTest member",
+        description: "All non-test XCTest members should be private.",
+        kind: .lint,
+        nonTriggeringExamples: NonPrivateXCTestMembersRuleExamples.nonTriggeringExamples,
+        triggeringExamples: NonPrivateXCTestMembersRuleExamples.triggeringExamples,
+        corrections: NonPrivateXCTestMembersRuleExamples.corrections
+    )
+
+    public func validate(file: SwiftLintFile) -> [StyleViolation] {
+        return violationRanges(in: file).map { range in
+            StyleViolation(ruleDescription: Self.description,
+                           severity: configuration.severity,
+                           location: Location(file: file, characterOffset: range.location))
+        }
+    }
+
+    // MARK: - SubstitutionCorrectableRule
+
+    public func violationRanges(in file: SwiftLintFile) -> [NSRange] {
+        return testClasses(in: file).flatMap { violationRanges(in: file, for: $0) }
+    }
+
+    public func substitution(for violationRange: NSRange, in file: SwiftLintFile) -> (NSRange, String)? {
+        return (violationRange, "private ")
```

Comment:
This has the potential of making code that previously compiled stop compiling in several cases:

1. If there was an explicit ACL keyword used
2. If the type was used either outside the file (internal) or outside the type scope (fileprivate)

I'm comfortable with those tradeoffs but let me know if you're not.